PIP_COMPATIBILITY.md

```diff
@@ -0,0 +1,277 @@
+
+This is a deliberate choice to avoid the correctness and security issues associated with allowing
+transitive dependencies to introduce arbitrary URLs into the dependency graph.
+
+For example:
+
+- Your package depends on `package_a==1.0.0`.
+- Your package depends on `package_b==1.0.0`.
+- `package_b==1.0.0` depends on `package_a @ https://...`.
+
+If `package_a @ https://...` happens to resolve to version `1.0.0`, `pip` would install `package_a`
+from the direct URL. This is a security issue, since the direct URL could be controlled by an
+attacker, and a correctness issue, since the direct URL could resolve to an entirely different
+package with the same name and version.
+
+In the future, uv may allow transitive URL dependencies in some form (e.g., with user opt-in).
+For more, see [#1808](https://github.com/astral-sh/uv/issues/1808).
+
+## Pre-release compatibility
+
+By default, uv will accept pre-release versions during dependency resolution in two cases:
+
+1. If the package is a direct dependency, and its version markers include a pre-release specifier
+   (e.g., `flask>=2.0.0rc1`).
+1. If _all_ published versions of a package are pre-releases.
+
+If dependency resolution fails due to a transitive pre-release, uv will prompt the user to
+re-run with `--prerelease=allow`, to allow pre-releases for all dependencies.
+
+Alternatively, you can add the transitive dependency to your `requirements.in` file with
+pre-release specifier (e.g., `flask>=2.0.0rc1`) to opt in to pre-release support for that specific
+dependency.
+
+In sum, uv needs to know upfront whether the resolver should accept pre-releases for a given
+package. `pip`, meanwhile, will generally respect pre-release identifiers in transitive
+dependencies.
+
+Pre-releases are [notoriously difficult](https://pubgrub-rs-guide.netlify.app/limitations/prerelease_versions)
+to model, and are a frequent source of bugs in packaging tools. Even `pip`, which is viewed as a
+reference implementation, has a number of open questions around pre-release handling ([#12469](https://github.com/pypa/pip/issues/12469),
+[#12470](https://github.com/pypa/pip/issues/12470), etc.). uv's pre-release handling is
+_intentionally_ limited and _intentionally_ requires user opt-in for pre-releases, to ensure
+correctness.
+
+In the future, uv _may_ support pre-release identifiers in transitive dependencies. However, it's
+likely contingent on evolution in the Python packaging specifications. The existing PEPs [do not
+cover "dependency resolution"](https://discuss.python.org/t/handling-of-pre-releases-when-backtracking/40505/17)
+and are instead focused on behavior for a _single_ version specifier. Further, the [original intent](https://discuss.python.org/t/handling-of-pre-releases-when-backtracking/40505/20)
+of the spec was _not_ to support cases in which pre-releases are _required_ for resolution despite
+a lack of pre-release specifiers. As such, there are unresolved questions around the correct and
```

Comment:
I'd skip the original intent part, back there dependency resolution wasn't a thing in python so PEP 440 doesn't account for it, and instead move the link to the pip behavior paragraph above.